src/order/well_founded.lean

```diff
@@ -137,6 +137,10 @@ section linear_order
 variables {β : Type*} [linear_order β] (h : well_founded ((<) : β → β → Prop))
   {γ : Type*} [partial_order γ]
 
+theorem min_le {x : β} {s : set β} (hx : x ∈ s) (hne : s.nonempty := ⟨x, hx⟩) :
+  h.min s hne ≤ x :=
```

Comment:
`hne` doesn't need to be an argument, due to proof irrelevancy you can just put `⟨x, hx⟩` in the theorem statement and it will work just fine.

Reply:
That's why `hne` is an argument with a default value. AFAIR, it may help if we have `h.min s hne` with some other proof (though two proofs should be equal by definition, so I'm not sure).

Reply:
Yeah, that's why I suggest just inlining the proof. There's no circumstance in which you need to provide both `hx` and `hne`.

Reply:
I think we've sometimes seen some weirdness where this behaves incorrectly due to `simp` or `rw` performing a syntactic match rather than unifying. This is definitely a pattern we've used elsewhere, but I don't think it ever got a library note to explain exactly why it matters.

Reply:
I've inlined a bunch of proofs into a bunch of statements, and I've never had any of those problems. I'm interested in seeing an example where Lean attempts to do syntactic matching on a proof.

Reply:
I've just met the following scenario. While proving `disjoint (s : set α) (t : set α)`, I did `rintro x ⟨hs, ht⟩` and got the goal `x ∈ ⊥`. Then I tried to write `exact (hd : disjoint s' t') ⟨_, _⟩` and got `x ∈ s'` as the first goal. I can't quickly create an example for this lemma.

Reply:
I'd be inclined to declare this out of scope, but maybe create a new issue about documenting this approach or cleaning it up